### ISSUE TITLE
Activity

### ISSUE THREAD

**Issue by Aza1** [DartVS/DartVS#96]
Is this still being actively developed?

**Comment by DanTup** [comment]
I took a break from dev on this due to health issues last year; and I've never been able to find time to get back to it. I also lost a lot of motivation after many weeks of wrestling with the Visual Studio extension APIs.

This repo has a lot of unfinished/unreleased work integrating with Google's Analyzer (a Dart service Google have written to make tooling integration easier and consistent across many IDEs) that I really wanted to get finished; but implementing turned out way more complicated/frustrating than I thought.
Although I'd love to have Dart support in VS, I just don't think it's something I can realistically deliver.

**Comment by Aza1** [comment]
Thanks for the update, and the effort.

There's definitely a need for this, even more so now with the Dart Editor's
demise.

**Comment by DanTup** [comment]
Yeah; I agree. I installed WebStorm yesterday but I don't think I'm going to get along with it very well :(

The Dart side of this integration is really simple (Google have done a great job with the Analysis Service), but the VS APIs are so frustrating (badly documented, many versions and MSDN doesn't make it clear which are preferred, full of out params and magic numbers, unclear which bits run on UI thread, silently swallow exceptions...) it's not a fun task. I've wasted weeks at a time trying to get basic project systems and code-completion working.

**Comment by nippur72** [comment]
would it be possible to separate the analysis server wrapper from the rest? I would like to explore the possibility to integrate Dart in [AvalonEdit](https://github.com/icsharpcode/AvalonEdit).

**Comment by DanTup** [comment]
It should be. Most (if not all) is already in a separate project, and there are some scripts for updating the Analysis Service snapshot (though this now ships with the SDK and might be better to use from there) and generating JSON classes from the docs.

I actually spend a little time updating things yesterday, and have had an offer of help from someone else with experience with VS extensions; so there's a possibility things might get back on track. I'd have no problem pulling the Analysis Service wrapper out into a NuGet package or something; but any time I have to spend on this for now will be trying to get the extension to a point of being able to release it. There's a lot of unreleased code in here that'd be really useful to people :(

**Comment by nippur72** [comment]
:+1:

**Comment by jrote1** [comment]
An idea would be to possibly develop this using roslyn api's. So would only be supported in visual studio 2015 but might be alot better api.

**Comment by DanTup** [comment]
Sadly; I've once again found little time to work on this :(

@jrote1 Roslyn is the .NET Compiler Platform, Dart doesn't really fit into it very well. The hard parts of parsing Dart are already done by Google anyway; I just piggybacked off that. You can use this directly (it's a Dart tool you run and communicate with over STDIN/STDOUT using JSON) if you're after building your own Dart tooling. See [here](https://groups.google.com/a/dartlang.org/forum/#!forum/analyzer-discuss).

**Comment by DanTup** [comment]
If it's of any interest, I recently started work on Dart support for **Visual Studio Code** (a lighter, cross-platform IDE from Microsoft). I've posted a preview release with basic functionality to get some feedback; feel free to try it out and let me know how you get on!

GH Link: https://github.com/DanTup/Dart-Code/releases/tag/v0.1.0
G+ Post: https://plus.google.com/113181962167438638669/posts/b75Fm31aFDJ
Dart-Misc Post: https://groups.google.com/a/dartlang.org/forum/#!topic/misc/xt0plnLFzkQ

**Comment by DanTup** [comment]
If it's of any interest, I recently started work on Dart support for Visual Studio Code (a lighter, cross-platform IDE from Microsoft). It's pre-v1 but has all of the basics (syntax highlighting, code completion, go to def, find references, search, formatting, running command line apps)! :)

https://marketplace.visualstudio.com/items?itemName=Dart-Code.dart-code

**Comment by nippur72** [comment]
Thanks, yes it's interesting VSCode is getting very popular, I myself use it as default text editor, I prefer it over Atom or other alternatives.

**Comment by DanTup** [comment]
Well if you're still writing Dart, I'd love to have [feedback](https://github.com/Dart-Code/Dart-Code/issues) on it; I'm only just getting back to Dart (and new to Code) so feedback from others is invaluable :)
